### PR TITLE
Added FormType to CCS Field Launch Endpoint

### PR DESCRIPTION
**Pull request by LeoKHoward**
# Motivation and Context
More easily identifies the FormType associated with the QID

# What has changed
QidDTO has been renamed to CCSFieldLaunchEndpointDTO
FormType has been added to DTO
Map has been created to specify which QID numbers relate to which FormType
CaseEndpoint IT and Unit tests have been updated to check for a FormType

# How to test?
Build and test

# Links
https://trello.com/c/MgjZzmX4
https://collaborate2.ons.gov.uk/confluence/display/SDC/ID%27s+2021
https://collaborate2.ons.gov.uk/confluence/display/CATD/Case+API

### CODE REVIEW

**Comment by AdamHawtin** [comment]
[Bug: Case Processors sample load QID generator mappings are wrong](https://trello.com/c/7IpafPL9/653-bug-case-processors-sample-load-qid-generator-mappings-are-wrong)

**Comment by LeoKHoward** [comment]
[Add Form type to CCS Field launch endpoint (3)](https://trello.com/c/MgjZzmX4/634-add-form-type-to-ccs-field-launch-endpoint-3)

File: src/main/java/uk/gov/ons/census/caseapisvc/model/dto/CCSFieldLaunchEndpointDTO.java [hunk 1]
```diff
@@ -3,7 +3,8 @@
 import lombok.Data;
 
 @Data
-public class QidDTO {
+public class CCSFieldLaunchEndpointDTO {
```
Comment (NickGrantX) [inline src/main/java/uk/gov/ons/census/caseapisvc/model/dto/CCSFieldLaunchEndpointDTO.java]: the name's a bit 'wordy' ya know? CCSLaunchDTO would have sufficed, for me personally.

File: src/main/java/uk/gov/ons/census/caseapisvc/utility/FormTypeHelper.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
+package uk.gov.ons.census.caseapisvc.utility;
+
+public class FormTypeHelper {
```
Comment (NickGrantX) [inline src/main/java/uk/gov/ons/census/caseapisvc/utility/FormTypeHelper.java]: This class duplicates functionality already found in `CaseEndpoint.mapQuestionnaireTypeToFormType` and it seems to add form types which will never be launched by CCS. CCS will only ever be doing household surveys, is my understanding - better check with Jenny.

File: src/main/java/uk/gov/ons/census/caseapisvc/utility/FormTypeHelper.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  private static final String HH_FORM_TYPE = "H";
+  private static final String IND_FORM_TYPE = "I";
+  private static final String CE1_FORM_TYPE = "C";
+  private static final String CONTINUATION = "Cont";
+
+  public static String mapCCSFieldLaunchQuestionnaireTypeToFormType(String qid) {
```
Comment (NickGrantX) [inline src/main/java/uk/gov/ons/census/caseapisvc/utility/FormTypeHelper.java]: Lovely descriptive function name (are you sure you weren't a programmer in a former life?) but we could lose the `FieldLaunch` party and it'd be a bit less wordy, but still just as descriptive. Also, see other comments about duplicated code and whether we _need_ all these mappings. Personally, I think something's gone wrong if we're launching a continuation questionnaire on EQ, for a CCS fieldworker.

**Review by NickGrantX — CHANGES_REQUESTED** [review]
Really good work, but a couple of minor comments that could do with being addressed.

File: src/main/java/uk/gov/ons/census/caseapisvc/utility/FormTypeHelper.java (hunk 3 above)
Comment (LeoKHoward) [reply src/main/java/uk/gov/ons/census/caseapisvc/utility/FormTypeHelper.java]: Can you look over the new changes and test it again please. Kept the map in but moved it to CaseEndpoint.java and removed the irrelevant FormTypes

File: src/main/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpoint.java [hunk 4]
```diff
@@ -119,13 +119,32 @@ public CaseContainerDTO findCaseByReference(
...
-    return qidDTO;
+
+    CCSLaunchDTO ccsLaunchDTO = new CCSLaunchDTO();
+    ccsLaunchDTO.setQuestionnaireId(ccsUacQidLink.getQid());
+    ccsLaunchDTO.setActive(ccsUacQidLink.isActive());
+    ccsLaunchDTO.setFormType(mapCCSQuestionnaireTypeToFormType(ccsUacQidLink.getQid()));
+    return ccsLaunchDTO;
+  }
+
+  public static String mapCCSQuestionnaireTypeToFormType(String qid) {
```
Comment (NickGrantX) [inline src/main/java/uk/gov/ons/census/caseapisvc/endpoint/CaseEndpoint.java]: shouldn't be public, shouldn't be static, and it should be down at the bottom of the class along with the other similar method.

**Review by NickGrantX — CHANGES_REQUESTED** [review]
Just one or two minor tweaks

**Review by NickGrantX — APPROVED** [review]
Looks great now. Good work

**Review by LukeLoze1 — APPROVED** [review]
Tests all pass.  I don't like everything being dumped in CaseEndPoint, but more than I like having to return FormType.  Anyway just an observation.

**Pull request merged**